source/postprocess/particle_density_histogram.cc

```diff
@@ -0,0 +1,268 @@
+                  TableIndices<dim> entry_index;
+                  entry_index[0] = x;
+                  entry_index[1] = y;
+                  //do another loop if in 3d
+                  if (dim == 3){
+                    for (unsigned int z=0; z<granularity; z++){                   
+                        entry_index[2] = z;
+                    }
+                  }                  
+                  double value_ideal = buckets_ideal(entry_index);
+                  double value_actual = static_cast<double>(buckets_actual(entry_index));
+                  actual_error_squared = (value_ideal - value_actual)*(value_ideal - value_actual);
+                }                    
+              }    
+              /*
+              take the ratio between the actual error and the worst case 
+              error, resulting in a score from 0 to 1 for the cell
+              */
+              double distribution_score_current_cell = actual_error_squared/worst_case_error_squared;
+              //average
+              global_score += distribution_score_current_cell;
+              //max
+              if (distribution_score_current_cell >local_max_score){
+                local_max_score = distribution_score_current_cell;
+              }
+              //min
+              if (distribution_score_current_cell < local_min_score){
+                local_min_score = distribution_score_current_cell;
+              }
+            }
+          }
+      }
+
+      //get final values from all processors
+      double global_max_score = Utilities::MPI::max (local_max_score, this->get_mpi_communicator());
+      double global_min_score = Utilities::MPI::min (local_min_score, this->get_mpi_communicator());
+      double summed_score = Utilities::MPI::sum (global_score, this->get_mpi_communicator());
+      double global_cells_with_particles = Utilities::MPI::sum (cells_with_particles, this->get_mpi_communicator());
+      double average_score = summed_score / global_cells_with_particles;
+
+      // write to statistics file
+      statistics.add_value ("Minimal density gradient score: ", global_min_score);
+      statistics.add_value ("Average density gradient score: ", average_score);
+      statistics.add_value ("Maximal density gradient score: ", global_max_score);
+
+      std::ostringstream output;
+      output << global_min_score << "/ " <<average_score << "/ " << global_max_score;
+
+      return std::pair<std::string, std::string> ("Particle density gradient score min/avg/max:",
+                                                  output.str());
+    }
+
+
+
+    template <int dim>
+    std::list<std::string>
+    ParticleDensityStatistics<dim>::required_other_postprocessors() const
+    {
+      return {"particles"};
+    }
+
+    
+    template <int dim>
+    void ParticleDensityStatistics<dim>::sortParticles(const typename Triangulation<dim>::active_cell_iterator &cell,Table<dim,unsigned int> &buckets,double bucket_width)
```

Comment:
split the parameters across separate lines
Also this function can be `const` as well I think.